src/aten/sycl/UpSampleBicubic2dKernels.cpp

```diff
@@ -0,0 +1,184 @@
+      int in_y = floorf(real_y);
+      accscalar_t t_y = real_y - in_y;
+      for (int n = 0; n < nbatch; n++) {
+        for (int c = 0; c < channels; c++) {
+          accscalar_t coefficients[4];
+          for (int k = 0; k < 4; k++) {
+            coefficients[k] = cubic_interp1d<scalar_t, accscalar_t>(
+                upsample_get_value_bounded<scalar_t>(
+                    idata,
+                    n,
+                    c,
+                    input_width,
+                    input_height,
+                    in_x - 1,
+                    in_y - 1 + k),
+                upsample_get_value_bounded<scalar_t>(
+                    idata,
+                    n,
+                    c,
+                    input_width,
+                    input_height,
+                    in_x + 0,
+                    in_y - 1 + k),
+                upsample_get_value_bounded<scalar_t>(
+                    idata,
+                    n,
+                    c,
+                    input_width,
+                    input_height,
+                    in_x + 1,
+                    in_y - 1 + k),
+                upsample_get_value_bounded<scalar_t>(
+                    idata,
+                    n,
+                    c,
+                    input_width,
+                    input_height,
+                    in_x + 2,
+                    in_y - 1 + k),
+                t_x);
+          }
+
+          odata[n][c][output_y][output_x] =
+              static_cast<scalar_t>(cubic_interp1d<scalar_t, accscalar_t>(
+                  coefficients[0],
+                  coefficients[1],
+                  coefficients[2],
+                  coefficients[3],
+                  t_y));
+        }
+      }
+    }
+  }
+  UpsampleBicubic2dOutFrameKernelFunctor(
+      PackedTensorAccessor64<scalar_t, 4> out_data,
+      const PackedTensorAccessor64<const scalar_t, 4> in_data,
+      int64_t onum,
+      bool align_corners,
+      const accscalar_t height_scale,
+      const accscalar_t width_scale)
+      : out_data_(out_data),
+        in_data_(in_data),
+        onum_(onum),
+        align_corners_(align_corners),
+        height_scale_(height_scale),
+        width_scale_(width_scale) {}
+
+ private:
+  PackedTensorAccessor64<scalar_t, 4> out_data_;
+  const PackedTensorAccessor64<const scalar_t, 4> in_data_;
+  int64_t onum_;
+  bool align_corners_;
+  const accscalar_t height_scale_;
+  const accscalar_t width_scale_;
+};
+
+template <typename scalar_t, typename accscalar_t>
+static void upsample_bicubic2d_out_frame(
+    PackedTensorAccessor64<scalar_t, 4> odata,
+    const PackedTensorAccessor64<const scalar_t, 4> idata,
+    int64_t onum,
+    bool align_corners,
+    const accscalar_t height_scale,
+    const accscalar_t width_scale) {
+  auto queue = getCurrentSYCLQueue();
+  int64_t local_range = static_cast<int64_t>(1);
+  int64_t global_range = static_cast<int64_t>(1);
+  if (onum != 0) {
```

Comment:
Why global range == 1, when onum == 0. This means we still launch kernel for onum==0?